ui/src/app/creditcard/creditcard.component.html

```diff
@@ -0,0 +1,77 @@
+<div class="creditcardPaymentOptions">
+  <button type="button" routerLink="/" routerLinkActive="active" class="btn btn-outline-secondary">Change Payment Type</button>
+</div>
+<br>
+<div>
+  <form #creditcardForm="ngForm">
+    <div class="container">
+      <div class="row">
+        <div class="col">
+          <div class="form-group creditcardLabels">
+            <label><b>Cards Accepted</b></label>
+          </div>
+        </div>
+        <div class="col">
+          <div class="form-group">
+            <label>Options</label>
+          </div>
+        </div>
+        <div class="w-100"></div>
+        <div class="col">
+          <div class="form-group creditcardLabels">
+            <label><b>Card Holder</b></label>
+          </div>
+        </div>
+        <div class="col">
+          <div class="form-group">
+            <input type="text" class="form-control" id="cardholder" required placeholder="enter your name"
+                   [(ngModel)]="creditCard.cardholder" name="cardholder" #cardHolder="ngModel">
+            <div [hidden]="cardHolder.valid" class="alert alert-danger">required</div>
```

Comment:
You could use Bootstrap's `invalid-feedback` indicators which creates a more subtle warning message for form validation.
 http://getbootstrap.com/docs/4.1/components/forms/#supported-elements

Reply:
These would have been sorted from having a design session. I will change it.

Reply:
Please provide me an example of a invalid-feedback or change one of the inputs on this code so I can see how they work. Tried for 30min to try and get them to work with no luck. Thank you

Reply:
Quick and easiest way is to create a decorator that will apply the styles bootstrap requires

```typescript
import {Directive, HostBinding, Self} from '@angular/core';
import { NgControl } from '@angular/forms';

@Directive({
	selector : '[formControlName],[ngModel],[formControl]',
})
export class BSControlStatusDirective {

	public constructor(@Self() private control: NgControl) {
	}

	@HostBinding('class.is-valid')
	get ngClassValid(): boolean {
		if (this.control.control == null) {
			return false;
		}
		return this.control.control.valid;
	}

	@HostBinding('class.is-invalid')
	get ngClassInvalid(): boolean {
		if (this.control.control == null) {
			return false;
		}
		return this.control.control.invalid;
	}
}
```
I use this in another project. It is adapted from this Stackoverflow example https://stackoverflow.com/questions/46810764/angular-4-bootstrap-4-consolidating-validation-classes (I cleaned up that example to match Angular's style guide https://angular.io/guide/styleguide#style-06-03)

Add that directive to the module, and it should automatically apply the styling for bootstrap.
You can then just add the `.invalid-feedback` or `.valid-feedback` items right after the input control like in the bootstrap examples http://getbootstrap.com/docs/4.1/components/forms/#how-it-works

This solves atleast
- Red or Green input fields depending on the validity
- Feedback messages if required

It does not solve the issue if you wanted a specific message for a specific validation error (e.g minlength, required, format, etc)